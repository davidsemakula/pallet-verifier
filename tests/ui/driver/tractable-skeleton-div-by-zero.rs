use std::marker::PhantomData;

pub trait Config {}

pub enum Call<T: Config> {
    __Ignore(PhantomData<T>),
    do_something { val: u8 },
}

pub struct Pallet<T: Config>(PhantomData<T>);

pub enum OriginFor<T> {
    Simple,
    Complex(T),
}

impl<T: Config> Pallet<T> {
    pub fn do_something(
        origin: OriginFor<T>,
        val: u8,
    ) {
        1 / val; //~ WARN: divide by zero
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    pub struct MyConfig;
    impl Config for MyConfig {}

    fn do_something_test() {
        let origin = OriginFor::Simple;
        Pallet::<MyConfig>::do_something(origin, 1);
    }
}