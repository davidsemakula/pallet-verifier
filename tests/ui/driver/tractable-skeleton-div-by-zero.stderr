warning: possible attempt to divide by zero
  --> tests/ui/driver/tractable-skeleton-div-by-zero.rs:25:13
   |
LL |             1 / val;
   |             ^^^^^^^

warning: 1 warning emitted

